subsys/bluetooth/services/ots/dir_list/ots_dir_list_lazy.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/dlist.h>
+#include <sys/byteorder.h>
+
+#include <bluetooth/services/ots.h>
+#include "../ots_internal.h"
+#include "../ots_obj_manager_internal.h"
+#include "ots_dir_list_internal.h"
+#include "ots_dir_list_impl_internal.h"
+
+#include <logging/log.h>
+
+LOG_MODULE_DECLARE(bt_ots, CONFIG_BT_OTS_LOG_LEVEL);
+
+/* In order to transmit the maximum size we choose CONFIG_BT_L2CAP_TX_MTU
+ * as the buffer size. However, directory listing encoding can only happen at
+ * directory entry boundary so we must ensure the MTU value is rounded up to the
+ * nearest directory entry size to not overflow our buffer. Also since we can only
+ * encode at object entry boundary, we need to make sure we have at least room for
+ * two entries so entries can be read from the middle.
+ */
+#define OTS_DIR_LIST_LAZY_BUFFER_SIZE MAX(2 * DIR_LIST_OBJ_RECORD_MAX_SIZE, \
+	ROUND_UP(CONFIG_BT_L2CAP_TX_MTU, DIR_LIST_OBJ_RECORD_MAX_SIZE))
```

Comment:
Why do we need the `ROUND_UP`? IIRC nothing in the directory listing is aligned to value in power of 2

Reply:
Yeah, I made a mistake here. I wanted to round up the value of MTU to a multiple of DIR_LIST_OBJ_RECORD_MAX_SIZE but this is the wrong way to do it.